src/flows/context/pipe.tsx

```diff
@@ -35,6 +38,12 @@ interface PipeContextProps {
 export const PipeProvider: FC<PipeContextProps> = ({id, children}) => {
   const {flow} = useContext(FlowContext)
   const results = useContext(ResultsContext)
+  const {generateMap} = useContext(QueryContext)
+
+  const stages = useMemo(() => generateMap(), [generateMap, flow.data.get(id)])
+  const queryText =
+    stages.filter(stage => stage.instances.includes(id))[0]?.text || ''
```

Comment:
lol i was about to yell about this being illegible, then remembered i told you to write it. so it looks perfect now